packages/block-editor/src/components/block-list/block.js

```diff
@@ -542,11 +469,10 @@ function BlockListBlock( {
 					) }
 				</Popover>
 			) }
-			<IgnoreNestedEvents
+			<div
 				ref={ blockNodeRef }
-				onDragStart={ preventDrag }
-				onMouseDown={ onMouseDown }
-				onMouseLeave={ onMouseLeave }
+				// Only allow selection to be started from a selected block.
+				onMouseLeave={ isSelected ? onMouseLeave : undefined }
```

Comment:
Is this related to multi-selection? And if it is, would any of it be expected to be handled in WritingFlow instead? Considering you mention it at https://github.com/WordPress/gutenberg/pull/19397/files#r363036970 as being partly responsible for handling multi-selection.

Reply:
That's right, this is for multi selection. It would be nice to move this to `WritingFlow` as well. I didn't do it here to avoid a huge PR, as it seems a little more tricky to solve.